### PR TITLE
Add all of the attributes broken down

### PR DESCRIPTION
**Pull request by trivigy**
This is a revamp of the sdp library which breaks down every single attribute into its own struct. Just a further enhancement of the library to adapt to the latest `pions/webrtc` demand.

### CODE REVIEW

**Comment by wdouglass** [comment]
This is a big commit! can we break it up into smaller ones? hard to review a whole-sale replacement of the library :smile:

File: candidate.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	"strconv"
+	"strings"
+
+	"github.com/pions/webrtc/pkg/rtcerr"
+
+	"github.com/pkg/errors"
+)
+
+// Candidate is defined in https://tools.ietf.org/html/rfc5245.
+type Candidate struct {
```
Comment (wdouglass) [inline candidate.go]: I guess this replaces the old `ICECandidate` structure -- how do I add a relatedaddress? i'm probably missing part of the spec

**Comment by trivigy** [comment]
@wdouglass For sure. I just don't have enough time to do it. That is what I was talking about in the chat. If you are up for it, feel free to just grab my code and chunk it into smaller PRs to get towards the things you need.

**Comment by wdouglass** [comment]
@trivigy cool, i'll try. Thanks again.

**Comment by trivigy** [comment]
If you break smaller PRs I am happy to do reviews/fixes/whatever along the way to help. Just cannot lead this change due to time constraints.

File: candidate.go (hunk 1 above)
Comment (trivigy) [reply candidate.go]: Oh year. I ended up completely separating it from the ICECandidate. When you talked about relatedaddress are you talking about the server reflexive address generation?

In general lets keep this sdp library separate from the webrtc and self contained. That means if there is need for a copy of a similar struct in both, it is most beneficial to do that. We can just do conversions inside of webrtc.

File: extmap_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExtmap(t *testing.T) {
+	tests := []struct {
+		parameter string
+		expected  string
+	}{
+		{exampleAttrExtmap1, exampleAttrExtmap1Line},
```
Comment (wdouglass) [inline extmap_test.go]: @trivigy did you forget to include these example strings?

Comment (trivigy) [reply extmap_test.go]: Nope. All there, they are just organized all in one file.

https://github.com/pions/sdp/pull/16/files#diff-0fe0a638de3620b2eaa11548e8c8ffcbR144

Comment (wdouglass) [reply extmap_test.go]: cool, missed that. Thanks.

**Comment by Sean-Der** [comment]
Hey @trivigy sorry this work didn't get merged :/

If you find yourself with sometime would love and get this merged eventually! Would be really great to have a type-safe and extensive SDP implementation. This is way better then reading the RFCs :)

**Comment by trivigy** [comment]
@Sean-Der it is okay. I really put this here not to be merged for so that someone else who needed pieces from it could add things down the stream. Unfortunately i really have no time doing this stuff these days.

**Pull request closed without merging**